src/dune

```diff
@@ -0,0 +1,48 @@
+(library
+  (name elpi)
+  (public_name elpi)
+  (preprocess (per_module
+    ((pps ppx_deriving.std trace_ppx --trace_ppx-on) elpi_runtime_trace_on)
+    ((pps ppx_deriving.std trace_ppx --trace_ppx-off) elpi_runtime_trace_off)
+    ((action (system "camlp5o -I . -I +camlp5 pa_extend.cmo pa_lexer.cmo %{input-file}")) elpi_parser)
+    ((pps ppx_deriving.std) elpi_util elpi_ast elpi_data elpi_compiler)))
+  (libraries re.str camlp5.gramlib unix)
+  (flags -linkall)
+  (modules elpi elpi_util elpi_parser elpi_ast elpi_compiler elpi_data elpi_ptmap elpi_builtin elpi_API elpi_runtime_trace_on elpi_runtime_trace_off)
+)
+
+(rule
+  (targets elpi_runtime_trace_on.ml)
+  (deps elpi_runtime.ml)
+  (action (copy# %{deps} %{targets}))
```

Comment:
Slightly shorter is to have:

```
(rule (copy# elpi_runtime.ml elpi_runtime_trace_on.ml))
```

Since dune is aware of how `copy#` works, it's able to infer what the dependencies and targets are already.

Reply:
thanks